portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -54,13 +55,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     switch(commentsAction){
       case "next":
-        jsonArray = comments.nextChunk(oldest, size);
+        if(language.equals("org"))
+          jsonArray = comments.nextChunk(oldest, size).asJson();
+        else
+          jsonArray = comments.nextChunk(oldest, size).asJson(language);
       break;
       case "prev":
-        jsonArray = comments.prevChunk(newest, size);
+        if(language.equals("org"))
+          jsonArray = comments.prevChunk(newest, size).asJson();
+        else
+          jsonArray = comments.prevChunk(newest, size).asJson(language);
       break;
       case "newest":
-        jsonArray = comments.newestChunk(size);
+        if(language.equals("org"))
+          jsonArray = comments.newestChunk(size).asJson();
+        else
+          jsonArray = comments.newestChunk(size).asJson(language);
```

Comment:
Instead of checking if language equals "org" here, can we check this within `asJson`, and let method `asJson` decide what special actions will apply if the language is "org".

Handling special cases at lower-level is cleaner as you don't need to duplicate these checks many times at higher-level call sites.

Reply:
Thanks, that sounds reasonable.